sql/catalyst/src/main/scala/org/apache/spark/sql/Encoder.scala

```diff
@@ -57,10 +57,6 @@ import org.apache.spark.sql.types._
  *   Encoders.bean(MyClass.class);
  * }}}
  *
- * == Implementation ==
- *  - Encoders are not required to be thread-safe and thus they do not need to use locks to guard
- *    against concurrent access if they reuse internal buffers to improve performance.
- *
```

Comment:
Removing this means Encoders must be thread-safe? Do we need explicit comment for that?

Reply:
I think it should be thread-safe.